pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -406,8 +407,32 @@ func newBlackholeFilter(enableAny bool) *listener.Filter {
 	return filter
 }
 
+func createRBACfilter(protocol plugin.ListenerProtocol, env *model.Environment, node *model.Proxy, push *model.PushContext) interface{} {
+	in := &plugin.InputParams{
+		Env:              env,
+		Node:             node,
+		Push:             push,
+		ListenerProtocol: protocol,
+	}
+	mutable := &plugin.MutableObjects{
+		FilterChains: []plugin.FilterChain{
+			{
+				ListenerProtocol: protocol,
+			},
+		},
+	}
+	if err := authz.NewPlugin().OnInboundListener(in, mutable); err == nil {
+		if protocol == plugin.ListenerProtocolTCP && len(mutable.FilterChains[0].TCP) == 1 {
+			return mutable.FilterChains[0].TCP[0]
```

Comment:
Do we need bounds checks here? This seems fragile?

Reply:
`mutable.FilterChains` is guaranteed to have at least 1 element from line 420